### ISSUE TITLE
Force update PureComponent on language change

### ISSUE THREAD

**Issue by vitexikora** [lingui/js-lingui#54]
Having these two translations of the same word in a component
```jsx
<input type="text" className="search__input" placeholder={i18n.t`Hledej`} />
<Trans>Hledej</Trans>
```
the placeholder is not updating correctly, while the usual `<Trans>` is. The component is updated upon language change, as well as the component with Provider. The placeholder is updated always on a second update (eg. a navigation), which causes it to be always in the wrong language when toggling languages (is delayed by 1 update). There is no problem with the `<Trans>` component.
Is there a way to resolve or bypass this problem (other than forcing update on didUpdate in every component...) ?
Thank you.

UPDATE:
Problem is resolved using Component instead of PureComponent, you are probably using context updates, which are ignored in PCs.

**Comment by tricoder42** [comment]
You're right, messages and active languages and propagated using context. However, there's a subscribe/broadcast mechanism to bypass these kind of problems and triggering `forceUpdate` automatically when language is changed or messages are loaded.

I'll take a look into it. I think it might be caused, that `WithI18n` triggers `forceUpdate` on itself, but underlying PureComponent still ignores it. I tested it only with translated component **inside** PureComponents (not PureComponent inside translated one).

**Comment by tricoder42** [comment]
I'm thinking about one possible workaround: every time active language is changed or message catalog updated, generate unique i18n-hash. `withI18n` will pass `i18n` object and `i18nHash` prop. This should work with PureComponent out of the box, as it does shallow comparison of props. 

The only drawback is, that component will receive one extra prop. Every prop has a risk of possible collision with user-defined props, but `i18nHash` should be relatively safe to use...

**Comment by tricoder42** [comment]
Fixed in **lingui-react@1.4.0**.

`i18nHash` is passed down from `withI18n` HOC by default. It can be disabled using `withHash = false` option (`withI18n({ withHash: false })`).

More info in [docs](https://lingui.github.io/js-lingui/guides/optimized-components.html)
